spec/requests/proposals_spec.rb

```diff
@@ -13,5 +13,104 @@
       expect(response.status).to eq(200)
     end
   end
-end
 
+  describe 'POST /proposals/:id/approve' do
+    def expect_status(proposal, status)
+      proposal.reload
+      proposal.approvals.each do |approval|
+        expect(approval.status).to eq(status)
+      end
+      expect(proposal.status).to eq(status)
+    end
+
+    before do
+      allow_any_instance_of(Proposal).to receive(:client).and_return('ACME')
+      allow_any_instance_of(Proposal).to receive(:public_identifier).and_return('123')
```

Comment:
For better or worse, I didn't want to have to create the associated model just to have these two delegated attributes available. Maybe I should just bite the bullet, short of doing https://www.pivotaltracker.com/story/show/93936068.